src/authhelper.cpp

```diff
@@ -0,0 +1,36 @@
+#include <QObject>
+#include <QProcess>
+
+#include <KAuth>
+#include <QDebug>
+#include <QEventLoop>
+#include <QTimer>
+
+class AuthHelper : public QObject
+{
+    Q_OBJECT
+public:
+    AuthHelper() = default;
+
+public slots:
+    KAuth::ActionReply elevate(const QVariantMap& args)
+    {
+        QEventLoop loop;
+        KAuth::ActionReply reply = KAuth::ActionReply::SuccessReply();
+
+        auto process = new QProcess();
+
+        QTimer::singleShot(1000, process, [process] {
```

Comment:
remove, and instead use `process.waitForFinished` with a timeout of 1s, return a failure reply if that doesn't work